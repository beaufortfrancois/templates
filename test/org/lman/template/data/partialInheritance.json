{
  "foo": {"bar": "baz"},
  "arg1": {"qux": 10},
  "arg2": {"qux": 20},
  "arg3": {"qux": 30},
  "deep": {
    "arg2": {"qux": 40},
    "arg3": {"qux": 50},
    "arg4": {"qux": 60},
    "moreDeep": {
      "arg3": {"qux": 70},
      "arg4": {"qux": 80},
      "arg5": {"qux": 90}
    }
  }
}
