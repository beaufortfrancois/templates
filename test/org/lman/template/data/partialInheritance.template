1. Partials should inherit the initial/global state of contexts:
2. Partials should either inherit the topmost-only local context,
   or some explicit context given by arguments.

 no args: {{+partials.partialInheritance_p1}}
arg:arg1: {{+partials.partialInheritance_p1 arg:arg1}}
arg:arg2: {{+partials.partialInheritance_p1 arg:arg2}}

{{#deep}}
 no args: {{+partials.partialInheritance_p1}}
arg:arg1: {{+partials.partialInheritance_p1 arg:arg1}}
arg:arg2: {{+partials.partialInheritance_p1 arg:arg2}}

{{#moreDeep}}
 no args: {{+partials.partialInheritance_p1}}
arg:arg1: {{+partials.partialInheritance_p1 arg:arg1}}
arg:arg2: {{+partials.partialInheritance_p1 arg:arg2}}
{{/moreDeep}}
{{/deep}}
