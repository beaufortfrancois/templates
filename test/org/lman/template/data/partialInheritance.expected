1. Partials should inherit the initial/global state of contexts:
2. Partials should either inherit the topmost-only local context,
   or some explicit context given by arguments.

 no args: foo.bar:baz, arg.qux:, arg1.qux:10, arg2.qux:20, arg3.qux:30, arg4.qux:, arg5.qux:.
arg:arg1: foo.bar:baz, arg.qux:10, arg1.qux:10, arg2.qux:20, arg3.qux:30, arg4.qux:, arg5.qux:.
arg:arg2: foo.bar:baz, arg.qux:20, arg1.qux:10, arg2.qux:20, arg3.qux:30, arg4.qux:, arg5.qux:.

 no args: foo.bar:baz, arg.qux:, arg1.qux:10, arg2.qux:40, arg3.qux:50, arg4.qux:60, arg5.qux:.
arg:arg1: foo.bar:baz, arg.qux:10, arg1.qux:10, arg2.qux:40, arg3.qux:50, arg4.qux:60, arg5.qux:.
arg:arg2: foo.bar:baz, arg.qux:40, arg1.qux:10, arg2.qux:40, arg3.qux:50, arg4.qux:60, arg5.qux:.

 no args: foo.bar:baz, arg.qux:, arg1.qux:10, arg2.qux:20, arg3.qux:70, arg4.qux:80, arg5.qux:90.
arg:arg1: foo.bar:baz, arg.qux:10, arg1.qux:10, arg2.qux:20, arg3.qux:70, arg4.qux:80, arg5.qux:90.
arg:arg2: foo.bar:baz, arg.qux:40, arg1.qux:10, arg2.qux:20, arg3.qux:70, arg4.qux:80, arg5.qux:90.
